pkg/graph/schema.graphql

```diff
@@ -2306,6 +2306,55 @@ enum TRBAdminNoteCategory {
   ADVICE_LETTER
 }
 
+"""
+Data specific to admin notes in the General Request category
+This type doesn't contain any actual data
+"""
+type TRBAdminNoteGeneralRequestCategoryData {
+  """
+  Placeholder field so this type is non-empty, always true
+  """
+  placeholderField: Boolean! # Necessary because gqlgen doesn't support empty types
+}
+
+"""
+Data specific to admin notes in the Initial Request Form category
+"""
+type TRBAdminNoteInitialRequestFormCategoryData {
+  appliesToBasicRequestDetails: Boolean!
+  appliesToSubjectAreas: Boolean!
+  appliesToAttendees: Boolean!
+}
+
+"""
+Data specific to admin notes in the Supporting Documents category
+"""
+type TRBAdminNoteSupportingDocumentsCategoryData {
+  documents: [TRBRequestDocument!]!
+}
+
+"""
+Data specific to admin notes in the Consult Session category
+This type doesn't contain any actual data
+"""
+type TRBAdminNoteConsultSessionCategoryData {
+  """
+  Placeholder field so this type is non-empty, always true
+  """
+  placeholderField: Boolean! # Necessary because gqlgen doesn't support empty types
+}
+
+"""
+Data specific to admin notes in the Advice Letter category
+"""
+type TRBAdminNoteAdviceLetterCategoryData {
+  appliesToMeetingSummary: Boolean!
+  appliesToNextSteps: Boolean!
+  recommendations: [TRBAdviceLetterRecommendation!]!
+}
+
+union TRBAdminNoteCategorySpecificData = TRBAdminNoteGeneralRequestCategoryData | TRBAdminNoteInitialRequestFormCategoryData | TRBAdminNoteSupportingDocumentsCategoryData | TRBAdminNoteConsultSessionCategoryData | TRBAdminNoteAdviceLetterCategoryData
```

Comment:
Would it make sense for category specific data to optionally be null instead of having a placeholder type? The category also lives on the parent note type which should let the FE know not to expect categorical data in those generic note cases, right?

Reply:
It might; I was hoping to use the approach of making `TRBAdminNote` a union type to get around this, but that didn't work, per [this comment](https://github.com/CMSgov/easi-app/pull/2231#issuecomment-1765408993). The backend code would be pretty similar either way. I'm not sure what would be more ergonomic for the frontend code.

Example FE code for working with query results, with the union type having cases/variants for all five categories (the current schema), showing exhaustive checking for all possible cases:
```ts
if (adminNote.categorySpecificData.__typename === 'TRBAdminNoteGeneralRequestCategoryData') {
  // do nothing, no actual data
} else if (adminNote.categorySpecificData.__typename === 'TRBAdminNoteInitialRequestFormCategoryData') {
  // Typescript now knows that adminNote.categorySpecificData has fields appliesToBasicRequestDetails, appliesToSubjectAreas, and appliesToAttendees
  // do something with those fields
} else if (adminNote.categorySpecificData.__typename === 'TRBAdminNoteSupportingDocumentsCategoryData' {
  // Typescript now knows that adminNote.categorySpecificData has a documents field
  // do something with those fields
} else if (adminNote.categorySpecificData.__typename === 'TRBAdminNoteConsultSessionCategoryData') {
  // do nothing, no actual data
} else if (adminNote.categorySpecificData.__typename === 'TRBAdminNoteAdviceLetterCategoryData') {
  // Typescript now knows that adminNote.categorySpecificData has fields appliesToMeetingSummary, appliesToNextSteps, recommendations
  // do something with those fields
}
```

Alternatively, if the GraphQL schema doesn't include the cases with no additional data, makes `adminNote.categorySpecificData` nullable, and defines `union TRBAdminNoteCategorySpecificData = TRBAdminNoteInitialRequestFormCategoryData | TRBAdminNoteSupportingDocumentsCategoryData | TRBAdminNoteAdviceLetterCategoryData`, the code might look like:
```ts
if (adminNote.categorySpecificData) {
  if (adminNote.categorySpecificData.__typename === 'TRBAdminNoteInitialRequestFormCategoryData') {
    // Typescript now knows that adminNote.categorySpecificData has fields appliesToBasicRequestDetails, appliesToSubjectAreas, and appliesToAttendees
    // do something with those fields
  } else if (adminNote.categorySpecificData.__typename === 'TRBAdminNoteSupportingDocumentsCategoryData' {
    // Typescript now knows that adminNote.categorySpecificData has a documents field
    // do something with those fields
  } else if (adminNote.categorySpecificData.__typename === 'TRBAdminNoteAdviceLetterCategoryData') {
    // Typescript now knows that adminNote.categorySpecificData has fields appliesToMeetingSummary, appliesToNextSteps, recommendations
    // do something with those fields
  }
}
```

This avoids the messiness of needing placeholder fields, but the FE code doesn't necessarily clearly show that it's covering all categories. @aterstriep, @adamodd, what do you think?

Reply:
Nullable `adminNote.categorySpecificData` looks neat to me, but either was is probably fine. Would `categorySpecificData` only need to show properties of one category data type? If so it's probably ok that FE can't see them all.

Reply:
@adamodd `categorySpecificData` would only show properties of one category's data at a time. Before checking `__typename`, no fields on it would be accessible; once the value of `__typename` was checked, Typescript would allow access to the fields of the data for that particular category.
The reason I've been thinking about having all five categories in the union type is essentially to be resilient to future changes, to make clear that the field could include data specific to any category. That might be overengineering it, though.

Reply:
Per @adamodd's comment [here](https://github.com/CMSgov/easi-app/pull/2231#issuecomment-1767646782), I'm keeping the union type covering all five categories (which means keeping the placeholder fields in place, though I've changed them to be nullable, and specified that they'll always be null in https://github.com/CMSgov/easi-app/pull/2231/commits/cc22b92431c70cb24d34eb27c4b9e06cee93f2de) and keeping `categorySpecificData` as non-nullable on the `TRBAdminNote` type.